Write navigation docs (#149)
mintlify/mint: docs/navigation.mdx

---
title: "Navigation"
description: "You need to update mint.json to add a page to the navigation menu"
---

## Navigation syntax

Our navigation syntax is recursive letting you make nested navigation groups. You don't need to include `.mdx` in page names.

<CodeGroup>

```json Regular Navigation
"navigation": [
    {
        "group": "Getting Started",
        "pages": ["quickstart"]
    }
]
```

```json Nested Navigation

> "navigation": [
>     {
>         "group": "Design Principles",
>         "pages": [
>             "overview",
>             {
>                 "group": "Nested Reference Pages",
>                 "pages": ["nested-page-name"]
>             }
>         ]
>     }

Reviewer: Keep this consistent with the regular one, so it's easy to see the differences between the two.
```suggestion
"navigation": [
    {
        "group": "Getting Started",
        "pages": [
            "quickstart",
            {
                "group": "CLI",
                "pages": ["cli/overview", "cli-commands"]
            }
        ]
    }
```

]
```

</CodeGroup>

## Folders

> Simply put your `.mdx` files in folders and update the paths in `mint.json`.

Reviewer: ```suggestion
Simply put your MDX files in folders and update the paths in `mint.json`.
```

> For example, `https://yoursite.com/my-folder/my-page` requires you to have a folder called `my-folder` containing an mdx file called `my-page.mdx`.

Reviewer: Nit: Capitalize MDX

<Warning>

You cannot use `api` for the name of a folder unless you nest it inside another folder. Mintlify uses Next.js which reserves the `api` folder for internal server calls.

Good: `other-folder/api/page-name`

Bad: `api/other-folder/page-name`

</Warning>

```json Navigation With Folder
"navigation": [
    {
        "group": "Group Name",
        "pages": ["my-folder/my-page"]
    }
]
```

## Hidden Pages

> Mdx files you don't include in `mint.json` will still show up on your website. Hidden pages never show up in the sidebar navigation but can be found through the search bar.

Reviewer: ```suggestion
MDX files you don't include in `mint.json` will not be included in the sidebar, but are accessible by navigating directly to the path or through the search bar.
```
